File: src/main/java/se/lu/nateko/cp/netcdf/viewing/Raster.java
package se.lu.nateko.cp.netcdf.viewing;

public interface Raster {

	/**
	 * @param lon longitude coordinate index
	 * @param lat latitude coordinate index
	 * @return the value
	 */
	double get(int lon, int lat);
	int getSizeLon();
	int getSizeLat();

	double getMin();
	double getMax();
	double getLatMin();
	double getLatMax();
	double getLonMin();
	double getLonMax();
	
	default double[][] to2DArray(){
		int nx = getSizeLon(), ny = getSizeLat();
		double[][] res = new double[ny][nx];
		for(int j = 0; j < ny; j++){
			double[] row = res[j];
			for(int i = 0; i < nx; i++){
				row[i] = get(i, j);
			}
		}
		return res;
	}

}

File: netcdf/src/main/scala/se/lu/nateko/cp/data/formats/netcdf/viewing/Raster.scala
package se.lu.nateko.cp.data.formats.netcdf.viewing

trait Raster {
	/**
	 * @param lon longitude coordinate index
	 * @param lat latitude coordinate index
	 * @return the value
	 */
	
	def get(lon: Int, lat: Int): Double
	def getSizeLon: Int
	def getSizeLat: Int

	def getMin: Double
	def getMax: Double

	def getLatMin: Double
	def getLatMax: Double

	def getLonMin: Double
	def getLonMax: Double

	def to2DArray: Array[Array[Double]] = Array.tabulate(getSizeLat, getSizeLon)((lat, lon) => get(lat, lon))
}
